### PR TITLE
improve(adapters): swap zksync adapter with generic adapter

### PR DESCRIPTION
**Pull request by bmzig**
Another switch like https://github.com/across-protocol/relayer/pull/1714

### CODE REVIEW

**Review by james-a-morris — COMMENTED** [review]
wow this is a huge amount of savings. LGTM

File: src/clients/bridges/ZKSyncAdapter.ts [hunk 1]
```diff
@@ -1,397 +1,32 @@
...
-      });
+    const { ZK_SYNC, MAINNET } = CHAIN_IDs;
+    const bridges = {};
+    const l2Signer = spokePoolClients[ZK_SYNC].spokePool.signer;
+    const l1Signer = spokePoolClients[MAINNET].spokePool.signer;
+    SUPPORTED_TOKENS[ZK_SYNC]?.map((symbol) => {
+      const l1Token = TOKEN_SYMBOLS_MAP[symbol].addresses[MAINNET];
+      const bridgeConstructor = CUSTOM_BRIDGE[ZK_SYNC]?.[l1Token] ?? CANONICAL_BRIDGE[ZK_SYNC];
+      bridges[l1Token] = new bridgeConstructor(ZK_SYNC, MAINNET, l1Signer, l2Signer, l1Token);
     });
```
Comment (pxrl) [inline src/clients/bridges/ZKSyncAdapter.ts:20]: The comments to the Linea adapter are probably also relevant here: https://github.com/across-protocol/relayer/pull/1714, subject to the eventual implementation in that adapter :+1:

**Review by pxrl — COMMENTED** [review]
Just to confirm, was this tested w/ rebalances for the supported tokens?

**Comment by bmzig** [comment]
Tested inventory tracking with DAI, WETH, USDT, USDC, and WBTC. All are good.

**Comment by bmzig** [comment]
Now can confirm that event tracking works for DAI, WBTC, WETH, USDC, and USDT.

**Review by pxrl — APPROVED** [review]

**Pull request merged**